test/library/keycloakidp.go

```diff
@@ -0,0 +1,374 @@
+package library
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	configv1client "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	routev1client "github.com/openshift/client-go/route/clientset/versioned/typed/route/v1"
+)
+
+func AddKeycloakIDP(
+	t *testing.T,
+	kubeconfig *rest.Config,
+) (idpURL, idpName string, cleanups []func()) {
+	kubeClients, err := kubernetes.NewForConfig(kubeconfig)
+	require.NoError(t, err)
+
+	routeClient, err := routev1client.NewForConfig(kubeconfig)
+	require.NoError(t, err)
+
+	configClient, err := configv1client.NewForConfig(kubeconfig)
+	require.NoError(t, err)
+
+	// configclients, err := configv1client.NewForConfig(kubeconfig)
+	// require.NoError(t, err)
+
+	nsName, keycloakHost, cleanup := deployPod(t, kubeClients, routeClient,
+		"keycloak",
+		"docker.io/jboss/keycloak:latest",
+		[]corev1.EnvVar{
+			// configure password for GitLab root user
+			{Name: "KEYCLOAK_USER", Value: "admin"},
+			{Name: "KEYCLOAK_PASSWORD", Value: "password"},
+		},
+		8080,
+		8443,
+		[]corev1.Volume{
+			{
+				Name: "certkeypair",
+				VolumeSource: corev1.VolumeSource{
+					Secret: &corev1.SecretVolumeSource{
+						SecretName: servingSecretName,
+					},
+				},
+			},
+		},
+		[]corev1.VolumeMount{
+			{
+				Name:      "certkeypair",
+				MountPath: "/etc/x509/https",
+				ReadOnly:  true,
+			},
+		},
+		corev1.ResourceRequirements{
+			Requests: corev1.ResourceList{
+				"cpu":    resource.MustParse("500m"),
```

Comment:
You might want to use more CPU for better bootup time. I usually use 1 vCPU.

The memory could be adjusted as well. For a 1 user installation probably 600-700 MB should be enough.

Reply:
Good to know. From me playing with this, I could already see Keycloak booting pretty fast (compared to GitLab from the other test, but GitLab ofc brings way more than just an OIDC), but I'll ajdust these.